src/components/directory-content.tsx

```diff
@@ -448,34 +445,40 @@ const DirectoryContent = () => {
         );
     };
 
-    const handleMousePosition = useCallback((coordinates, isEmpty) => {
-        if (isEmpty) {
-            return {
-                mouseX: coordinates.right,
-                mouseY: coordinates.top + 25 * constants.VERTICAL_SHIFT,
-            };
-        } else {
-            return {
-                mouseX: coordinates.left,
-                mouseY: coordinates.bottom,
-            };
-        }
-    }, []);
+    const handleMousePosition = useCallback(
+        (coordinates: DOMRect, isEmpty: boolean): { mouseX: number | null; mouseY: number | null } => {
+            if (isEmpty) {
+                return {
+                    mouseX: coordinates.right,
+                    mouseY: coordinates.top + 25 * constants.VERTICAL_SHIFT,
+                };
+            } else {
+                return {
+                    mouseX: coordinates.left,
+                    mouseY: coordinates.bottom,
+                };
+            }
+        },
+        []
+    );
 
     const handleDialog = useCallback(
-        (mouseEvent, isEmpty) => {
-            const coordinates = mouseEvent.target.getBoundingClientRect();
+        (mouseEvent: React.MouseEvent<HTMLElement>, isEmpty: boolean) => {
+            const coordinates: DOMRect = (mouseEvent.target as HTMLElement).getBoundingClientRect();
             //set the contextualMenu position
             setMousePosition(handleMousePosition(coordinates, isEmpty));
             setOpenDirectoryMenu(true);
 
-            dispatch(setActiveDirectory(selectedDirectory.elementUuid));
+            dispatch(setActiveDirectory(selectedDirectory?.elementUuid));
         },
         [dispatch, selectedDirectory?.elementUuid, handleMousePosition]
     );
 
     const renderEmptyDirContent = () => (
-        <EmptyDirectory openDialog={(mouseEvent) => handleDialog(mouseEvent, true)} theme={selectedTheme} />
+        <EmptyDirectory
+            openDialog={(mouseEvent: React.MouseEvent<HTMLElement>) => handleDialog(mouseEvent, true)}
```

Comment:
you don't need to explicitly put the type, it can be infered
```suggestion
            openDialog={(mouseEvent) => handleDialog(mouseEvent, true)}
```

Reply:
Ok